tsmo/src/modified_spat_plotter.py

```diff
@@ -0,0 +1,97 @@
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import math
+import matplotlib.pyplot as plt
+from matplotlib.pyplot import figure
+import math
+import constants
+import sys
+import os
+from matplotlib.ticker import (MultipleLocator, FormatStrFormatter, AutoMinorLocator)
+import datetime as dt
+import matplotlib.dates as md
+pd.options.mode.chained_assignment = None
+
+def modified_spat_plotter():
+    input_directory_path = f'{constants.DATA_DIR}/{constants.PARSED_OUTPUT_DIR}'
+    output_directory_path = f'{constants.DATA_DIR}/{constants.PLOT_DIR}'
+   
+    all_in_filenames = os.listdir(input_directory_path)
+    for file in all_in_filenames:
+        #read the parsed spat data file generated by the modified_spat_parser script
+        if "modified_spat" in file and "parsed" in file:
+            modified_spat_data = pd.read_csv(f'{input_directory_path}/{file}')
+            signal_groups = modified_spat_data['Signal_Group'].unique()
+
+            #Use moy and timestamp fields to get epoch time for each record
+            first_day_epoch = dt.datetime(2022, 1, 1, 0, 0, 0).timestamp() * 1000 #get time since epoch for beggining of year in ms
```

Comment:
This script will break when the year changes. You may want to retrieve the current year.